meilisearch/index.py

```diff
@@ -1530,8 +1544,13 @@ def __settings_url_for(self, sub_route: str) -> str:
     def _build_url(
         self,
         primary_key: Optional[str] = None,
+        csv_delimiter: Optional[str] = None,
     ) -> str:
-        if primary_key is None:
+        parameters = {}
+        if primary_key:
+            parameters["primaryKey"] = primary_key
+        if csv_delimiter:
+            parameters["csvDelimiter"] = csv_delimiter
+        if primary_key is None and csv_delimiter is None:
```

Comment:
I don't like that much to see these `if` conditions because they usually hide a code that will grow indefinitely, WDYT about this?

```suggestion
        parameters = { "csvDelimiter": csv_delimiter, "primaryKey": primary_key }
        parameters = dict((k, v) for k, v in parameters.items() if v)

        if primary_key is None and csv_delimiter is None:
```

Reply:
This will work, but I would argue it is less readable and not obvious what you are doing. If you do go this direction you could clean it up some:

```py
parameters = {k: v for k, v in parameters.items() if v}
```

With either direction it could also be worth it to short circuit so parameters don't get created and run if not needed:


```py
    def _build_url(
        self,
        primary_key: Optional[str] = None,
        csv_delimiter: Optional[str] = None,
    ) -> str:
        if primary_key is None and csv_delimiter is None:
            return f"{self.config.paths.index}/{self.uid}/{self.config.paths.document}"

           parameters = { "csvDelimiter": csv_delimiter, "primaryKey": primary_key }
           parameters = {k: v for k, v in parameters.items() if v}

        return f"{self.config.paths.index}/{self.uid}/{self.config.paths.document}?{parse.urlencode(parameters)}"

```